12-connectivity.tf

```
@@ -2,73 +2,17 @@
 # Connectivity
 #--------------------------------------------------------------
 
-# Subnets
+## Subnets
 
-## AKS-00
-
-resource "azurerm_subnet" "aks_00_subnet" {
-  address_prefixes = [var.aks_00_subnet_cidr_blocks]
-
-  name = format("%s-00",
-    var.service_shortname
-  )
-
-  resource_group_name  = var.resource_group_name
-  virtual_network_name = azurerm_virtual_network.virtual_network.name
-  service_endpoints    = var.subnet_service_endpoints
-}
-
-## AKS-01
-
-resource "azurerm_subnet" "aks_01_subnet" {
-  address_prefixes = [var.aks_01_subnet_cidr_blocks]
-
-  name = format("%s-01",
-    var.service_shortname
-  )
-
-  resource_group_name  = var.resource_group_name
-  virtual_network_name = azurerm_virtual_network.virtual_network.name
-  service_endpoints    = var.subnet_service_endpoints
-
-}
-
-## Iaas 
-
-resource "azurerm_subnet" "iaas_subnet" {
-  address_prefixes = [var.iaas_subnet_cidr_blocks]
-
-  name = "iaas"
-
-  resource_group_name                            = var.resource_group_name
-  virtual_network_name                           = azurerm_virtual_network.virtual_network.name
-  service_endpoints                              = var.subnet_service_endpoints
-  enforce_private_link_endpoint_network_policies = var.iaas_subnet_enforce_private_link_endpoint_network_policies
-}
-
-## Application Gateway
-
-resource "azurerm_subnet" "application_gateway_subnet" {
-  address_prefixes = [var.application_gateway_subnet_cidr_blocks]
-
-  name = format("%s-appgw",
-    var.service_shortname
-  )
-
-  resource_group_name  = var.resource_group_name
-  virtual_network_name = azurerm_virtual_network.virtual_network.name
-}
-
-## VH Additional Subnets
-
-resource "azurerm_subnet" "additional_subnets" {
-  for_each = { for subnet in var.additional_subnets : subnet.name => subnet }
+resource "azurerm_subnet" "subnets" {
+  for_each = { for subnet in var.subnets : subnet.name => subnet }
 
   name                                           = each.value.name
   address_prefixes                               = [each.value.address_prefix]
   resource_group_name                            = var.resource_group_name
   virtual_network_name                           = azurerm_virtual_network.virtual_network.name
-  enforce_private_link_endpoint_network_policies = true
+  enforce_private_link_endpoint_network_policies = each.value.enforce_private_link_endpoint_network_policies == null ? false : var.iaas_subnet_enforce_private_link_endpoint_network_policies
```

why are you using the `iaas_subnet_enforce_private_link_endpoint_network_policies` var? 

don't you want this?

```suggestion
  enforce_private_link_endpoint_network_policies = each.value.enforce_private_link_endpoint_network_policies == null ? false : each.value.enforce_private_link_endpoint_network_policies
```

🚫 Blocking

---

Was just grabbing it from the default defined in the module. Can remove and just include in my for_each.

---

Added now to each.value